src/replication/DealReplicationService.ts

```diff
@@ -81,31 +82,51 @@ export default class DealReplicationService extends BaseService {
         status: found.status,
         cronSchedule: found.cronSchedule,
         cronMaxDeals: found.cronMaxDeals,
-        cronMaxPendingDeals: found.cronMaxPendingDeals
+        cronMaxPendingDeals: found.cronMaxPendingDeals,
+        fileListPath: found.fileListPath,
+        notes: found.notes
       };
+      const count = request.query['count'];
+      if (count === 'true') {
+        result.dealsTotal = await Datastore.DealStateModel.count({ replicationRequestId: found.id });
```

Comment:
please use one aggregate command to get number of deals per status
https://github.com/tech-greedy/singularity/blob/main/src/deal-preparation/handler/GetPreparationRequestHandler.ts#L17